tedana/workflows/t2smap.py

```diff
@@ -21,48 +22,80 @@ def t2smap(data, tes, combmode='t2s', label=None):
+        'ts' means that the model is fit, per voxel and per timepoint.
+        Default is 'all'.
     combmode : {'t2s', 'ste'}, optional
         Combination scheme for TEs: 't2s' (Posse 1999, default), 'ste' (Poser).
     label : :obj:`str` or :obj:`None`, optional
         Label for output directory. Default is None.
     """
-    if label is not None:
-        suf = '_%s' % str(label)
-    else:
-        suf = ''
-    tes, data, combmode = tes, data, combmode
-
+    # ensure tes are in appropriate format
     tes = [float(te) for te in tes]
     n_echos = len(tes)
 
-    catd = utils.load_data(data, n_echos=n_echos)
-    _, n_echos, _ = catd.shape
+    # coerce data to samples x echos x time array
+    LGR.info('Loading input data: {}'.format([f for f in data]))
+    catd, ref_img = utils.load_data(data, n_echos=n_echos)
+    n_samp, n_echos, n_vols = catd.shape
+    LGR.debug('Resulting data shape: {}'.format(catd.shape))
 
-    ref_img = data[0] if isinstance(data, list) else data
+    try:
+        ref_label = os.path.basename(ref_img).split('.')[0]
+    except TypeError:
+        ref_label = os.path.basename(str(data[0])).split('.')[0]
+
+    if label is not None:
+        out_dir = 'TED.{0}.{1}'.format(ref_label, label)
+    else:
+        out_dir = 'TED.{0}'.format(ref_label)
+    out_dir = op.abspath(out_dir)
+    if not op.isdir(out_dir):
+        LGR.info('Creating output directory: {}'.format(out_dir))
+        os.mkdir(out_dir)
+    else:
+        LGR.info('Using output directory: {}'.format(out_dir))
 
     LGR.info('Computing adaptive mask')
     mask, masksum = utils.make_adaptive_mask(catd, minimum=False, getsum=True)
     utils.filewrite(masksum, 'masksum%s' % suf, ref_img, copy_header=False)
 
     LGR.info('Computing adaptive T2* map')
-    t2s, s0, t2ss, s0vs, _, _ = model.t2sadmap(catd, tes, mask, masksum, 2)
-    utils.filewrite(t2ss, 't2ss%s' % suf, ref_img, copy_header=False)
-    utils.filewrite(s0vs, 's0vs%s' % suf, ref_img, copy_header=False)
+    if fitmode == 'all':
+        (t2s_limited, s0_limited,
+         t2ss, s0s,
+         t2s_full, s0_full) = model.fit_decay(catd, tes, mask,
+                                              masksum, start_echo=1)
+    else:
+        (t2s_limited, s0_limited,
+         t2s_full, s0_full) = model.fit_decay_ts(catd, tes, mask, masksum,
+                                                 start_echo=1)
+
+    # set a hard cap for the T2* map/timeseries
+    # anything that is 10x higher than the 99.5 %ile will be reset to 99.5 %ile
+    cap_t2s = stats.scoreatpercentile(t2s_limited.flatten(), 99.5,
+                                      interpolation_method='lower')
+    LGR.debug('Setting cap on T2* map at {:.5f}'.format(cap_t2s * 10))
+    t2s_limited[t2s_limited > cap_t2s * 10] = cap_t2s
 
     LGR.info('Computing optimal combination')
-    tsoc = np.array(model.make_optcom(catd, t2s, tes, mask, combmode),
```

Comment:
Can you explain these changes for me ?

Reply:
Most of these changes are just taken from the current version of the `tedana` workflow, except for changing the output variable names from the `fit_decay` functions, which I did because I think the new ones are more informative. I also dropped the part where it writes out the `t2ss` and `s0vs` maps because `fit_decay_ts` doesn't return those and I honestly have no clue what they are.

Are there any changes in particular that don't make sense?

Reply:
Yes, sorry, that wasn't a helpful question on my part ! Your explanation solved it anyway, though: I was confused why we were renaming `tsoc` to `OCcatd`.